extensions/amp-analytics/0.1/test/vendor-requests.json

```diff
@@ -261,6 +261,10 @@
     "pageload": "https://engagement-collector.mobify.net/s.gif?slug=mobify-project-id&timestamp_local=_timestamp_&channel=web&dimensions=%7b%22platform%22%3a%22AMP%22%2c%22client_id%22%3a%22_client_id_%22%2c%22title%22%3a%22_title_%22%2c%22location%22%3a%22_source_url_%22%2c%22page%22%3a%22_source_path_%22%2c%22src_location%22%3a%22_ampdoc_url_%22%2c%22referrer%22%3a%22_document_referrer_%22%2c%22templateName%22%3a%22page-type%22%7d&data=%7b%22category%22%3a%22timing%22%2c%22action%22%3a%22load%22%2c%22value%22%3a_page_load_time_%7d",
     "pagedcl": "https://engagement-collector.mobify.net/s.gif?slug=mobify-project-id&timestamp_local=_timestamp_&channel=web&dimensions=%7b%22platform%22%3a%22AMP%22%2c%22client_id%22%3a%22_client_id_%22%2c%22title%22%3a%22_title_%22%2c%22location%22%3a%22_source_url_%22%2c%22page%22%3a%22_source_path_%22%2c%22src_location%22%3a%22_ampdoc_url_%22%2c%22referrer%22%3a%22_document_referrer_%22%2c%22templateName%22%3a%22page-type%22%7d&data=%7b%22category%22%3a%22timing%22%2c%22action%22%3a%22DOMContentLoaded%22%2c%22value%22%3a_content_load_time_%7d"
   },
+  "moengage": {
+    "addDevice": "https://websdk.moengage.com/v2/device/add?os=_os_&app_id=!appId&sdk_ver=_sdk_ver_&app_ver=_app_ver_&device_tz_offset=_device_tz_offset_&device_tz=_device_tz_&os_platform=_os_platform_&unique_id=_unique_id_&isAmp=true",
+    "event": "https://websdk.moengage.com/v2/report/add?os=_os_&app_id=!appId&sdk_ver=_sdk_ver_&app_ver=_app_ver_&device_tz_offset=_device_tz_offset_&device_tz=_device_tz_&os_platform=_os_platform_&unique_id=_unique_id_"
+  },
```

Comment:
"addDevice": "https://websdk.moengage.com/v2/device/add?os=mweb&app_id=!appId&sdk_ver=0.0.0&app_ver=1&device_tz=_timezone_&os_platform=_user_agent_&unique_id=_client_id(moe_uuid)_&device_ts=_timestamp_&isAmp=true&a=%5Bobject%20Object%5D&meta=%5Bobject%20Object%5D&url=_ampdoc_url_"

"event": "https://websdk.moengage.com/v2/device/add?os=mweb&app_id=!appId&sdk_ver=0.0.0&app_ver=1&device_tz=_timezone_&os_platform=_user_agent_&unique_id=_client_id(moe_uuid)_&device_ts=_timestamp_&a=%5Bobject%20Object%5D&meta=%5Bobject%20Object%5D&url=_ampdoc_url_"

Please review these corrections to ensure they're formatted the way you want and let me know if you have any questions. Note: the `extraUrlParams` that are objects get resolved to [object Object] in these tests, but since you have `useBody`, they will not resolve this way when the requests are actually sent.

Reply:
@micajuine-ho why it is resolving to the url params? Since I am using `useBody` and `xhrpost`, `extraUrlParams` should go in the request body.
If that is not the case, then how can I have objects in `extraUrlParams`?

Reply:
It will go in the request body in the actual request.  What's happening here is that our testing framework doesn't account for this scenario right now, and so it always appends the `extraUrlParams`

Reply:
Thanks for clarifying. What will you suggest me to do now? we are not in the situation to have this API rewritten from the backend.

Reply:
I can fix the testing setup to account for this situation correctly, and then we can retest to make sure that it aligns with your backend endpoints.

How does that sound?

Reply:
sounds great. Thanks a lot..
Hope it will take much time. :)

Reply:
@micajuine-ho Hi Micajuine, any update on this?

Reply:
Hi, no updates as of yet. We've had other items of high priority to attend to recently. I will try to carve some time out this week to finish this. Thanks for the reminder.